Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -2767,6 +2767,14 @@ TextStream& operator<<(TextStream& ts, const WebProcessProxy& process)
     return ts;
 }
 
+void WebProcessProxy::validatePublicSuffix(const String& publicSuffix)
+{
+    bool isPublicSuffix = PublicSuffixStore::singleton().isPublicSuffix(publicSuffix);
+    if (!isPublicSuffix)
+        WEBPROCESSPROXY_RELEASE_LOG_ERROR(Network, "handleUnrecognizedPublicSuffix: Received an recognized public suffix %" PUBLIC_LOG_STRING, publicSuffix.utf8().data());
```

Comment:
Isn't this going to log domains on customer builds?

Reply:
This should be TLD (`com`, `net`, etc) not eTLD + 1 (`apple.com`), so I don't think it is privacy sensitive. 
I can change it to private if you think it should be.

Reply:
What is the purpose of this patch? Why are we trying to validate the public suffix at all?

The code does this:
1. UIProcess sends public suffix to the WebProcess
2. Before adding to PublicSuffixStore, WebProcess checks if public suffix is valid
3. If it is not valid, it asks the UIProcess to validate it

Seems silly. Why ask the UIProcess to validate something it sent us?

Reply:
The assertion was introduced in [276834@main](https://commits.webkit.org/276834@main) to help debug rdar://125417343, where we suspected the cause is web process adding unexpected item to public suffix cache. 
There was a bug where UI sends topPrivatelyControlledDomain instead of publicSuffix [276635@main](https://commits.webkit.org/276635@main).

This patch aims to replace the assertion in web process. If we believe everything works correctly, then indeed we don't need to validation process.

Reply:
Thinking about it, maybe we could wrap the public suffix in a class and send that instead of a plain string in message instead. Do you think it's better to do this?

Reply:
Yes, having a PublicSuffix class makes sense to me.

Reply:
If you want I can do that too, if you want to go back to site isolation.

Reply:
Seems not too complained; I will work on it